### PR TITLE
fix: remove permission for Guest to list users

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:
fix: remove permission for Guest to list users

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?
- [ ] If new dependencies have been added, have they been pinned to specific versions?
- [ ] Is this change also required on the AWS Solution version?
- [ ] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)?
- [ ] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)?
- [ ] If you are updating the changelog and vending out a new release, have you updated versionNumber and versionDate in [.defaults.yml](../main/config/settings/.defaults.yml)

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by kpark277 — COMMENTED** [review]
Does this break any e2e tests? So guests can still reach the Users tab, but they will just get an empty table?

**Review by kpark277 — DISMISSED** [review]

File: addons/addon-base/packages/services/lib/user/user-authz-service.js [hunk 1]
```diff
@@ -49,6 +51,13 @@ class UserAuthzService extends Service {
   }
 
   // Protected methods
+  async authorizeList(requestContext) {
+    if (requestContext.principal.userRole === 'guest') {
```
Comment (Bingjiling) [inline addons/addon-base/packages/services/lib/user/user-authz-service.js]: Is there a differentiation between internal v.s. external guest here?

Comment (SanketD92) [reply addons/addon-base/packages/services/lib/user/user-authz-service.js]: Yes, both are different. @nguyen102 should we add `internal guest` here as well?

Comment (nguyen102) [reply addons/addon-base/packages/services/lib/user/user-authz-service.js]: Good catch. I updated the code to deny access for internal-guest as well

**Review by SanketD92 — APPROVED** [review]

**Pull request merged**